plugin/storage/memory/memory.go

```diff
@@ -339,3 +335,78 @@ func flattenTags(span *model.Span) model.KeyValues {
 	}
 	return retMe
 }
+
+func hasCorrespondingServerSpan(clientSpan *model.Span, spans []*model.Span) bool {
+	for _, span := range spans {
+		if span.ParentSpanID() == clientSpan.SpanID {
+			return true
+		}
+	}
+	return false
+}
+
+func isClientSpan(span *model.Span) bool {
+	for _, tag := range span.Tags {
+		if tag.Key == "span.kind" && tag.VStr == "client" {
+			return true
+		}
+	}
+	return false
+}
+
+func updateServiceDependencyLinks(parentService, childService string, deps map[string]*model.DependencyLink) {
+	depKey := parentService + "&&&" + childService
+	if _, ok := deps[depKey]; !ok {
+		deps[depKey] = &model.DependencyLink{
+			Parent:    parentService,
+			Child:     childService,
+			CallCount: 1,
+		}
+	} else {
+		deps[depKey].CallCount++
+	}
+}
+
+func inferServiceName(span *model.Span) string {
+    serviceName := "inferred"
+
+    // Check for peer.service tag
+    if peerService, found := getTagValue(span, "peer.service"); found {
+        return serviceName + "-" + peerService
+    }
+
+    // Check for RPC service name
+    if rpcService, found := getTagValue(span, "rpc.service"); found {
+        return serviceName + "-rpc-" + rpcService
+    }
+
+    // Check for HTTP service name via route or URL
+    if httpRoute, found := getTagValue(span, "http.route"); found {
+        return serviceName + "-http-" + httpRoute
+    } else if httpURL, found := getTagValue(span, "http.url"); found {
+        return serviceName + "-http-" + httpURL 
+    }
+
+    // Check for Database service name via DB name or connection string
+    if dbSystem, found := getTagValue(span, "db.system"); found {
+        dbName := dbSystem
+        if dbName == "" {
+            dbName = "unknown"
+        }
+        if dbStatement, found := getTagValue(span, "db.statement"); found {
```

Comment:
db.statement has no relation to service name

Reply:
Addressed!